web/api/v1/api.go

```diff
@@ -1479,6 +1443,18 @@ func (api *API) respondError(w http.ResponseWriter, apiErr *apiError, data inter
 	}
 }
 
+func parseTimeParam(r *http.Request, paramName string, defaultValue time.Time) (int64, time.Time, error) {
+	val := r.FormValue(paramName)
+	if val != "" {
+		result, err := parseTime(val)
+		if err != nil {
+			return -1, time.Time{}, errors.Wrapf(err, "Invalid parameters '%s'", paramName)
+		}
+		return timestamp.FromTime(result), result, nil
+	}
+	return timestamp.FromTime(defaultValue), defaultValue, nil
```

Comment:
it looks like this would call timestamp.FromTime even when we don't need it.

Reply:
Yeah, I know....just I don't want to provide an additional flag for this. Any ideas of how this can be improved?
Btw this could be a good ```golang``` feature - lazy loading return values :)

Reply:
We only use timestamp.FromTime once, so you can still use it externally.

Reply:
Actually is used 4 times - in deleteSeries, in series and + 2 times more when is combined with this PR https://github.com/prometheus/prometheus/pull/6837